### PR TITLE
Update links and code examples in the guides to use HTTPS

### PR DESCRIPTION
**Pull request by thehonkinggoose**
### Summary

Since many of the guide articles were initially written, HTTPS adoption on the internet has skyrocketed. Likewise many hosts that used to not, now do support HTTPS. This patch goes through links in the guides and changes them to use HTTPS.

Additionally if in a code example in a guide article references a website with http that supports https, the link inside the code example was changed to use HTTPS.

### Other Information

N/A

### CODE REVIEW

File: guides/source/i18n.md [hunk 1]
```diff
@@ -222,7 +222,7 @@ This approach has almost the same set of advantages as setting the locale from t
...
 
-Rails contains infrastructure for "centralizing dynamic decisions about the URLs" in its [`ApplicationController#default_url_options`](http://api.rubyonrails.org/classes/ActionDispatch/Routing/Mapper/Base.html#method-i-default_url_options), which is useful precisely in this scenario: it enables us to set "defaults" for [`url_for`](http://api.rubyonrails.org/classes/ActionDispatch/Routing/UrlFor.html#method-i-url_for) and helper methods dependent on it (by implementing/overriding `default_url_options`).
+Rails contains infrastructure for "centralizing dynamic decisions about the URLs" in its [`ApplicationController#default_url_options`](s/classes/ActionDispatch/Routing/Mapper/Base.html#method-i-default_url_options), which is useful precisely in this scenario: it enables us to set "defaults" for [`url_for`](https://api.rubyonrails.org/classes/ActionDispatch/Routing/UrlFor.html#method-i-url_for) and helper methods dependent on it (by implementing/overriding `default_url_options`).
```
Comment (sikachu) [inline guides/source/i18n.md]: This one has a typo.

```suggestion
Rails contains infrastructure for "centralizing dynamic decisions about the URLs" in its [`ApplicationController#default_url_options`](https://api.rubyonrails.org/classes/ActionDispatch/Routing/Mapper/Base.html#method-i-default_url_options), which is useful precisely in this scenario: it enables us to set "defaults" for [`url_for`](https://api.rubyonrails.org/classes/ActionDispatch/Routing/UrlFor.html#method-i-url_for) and helper methods dependent on it (by implementing/overriding `default_url_options`).
```

File: guides/source/routing.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**
+******DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**
```
Comment (sikachu) [inline guides/source/routing.md]: Would you mind undo this change?

```suggestion
**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**
```

**Review by sikachu — CHANGES_REQUESTED** [review]
I found some typos. Would you mind fixing them so I can merge in?

Thank you for the patch!

**Comment by vipulnsward** [comment]
Hi @nsuchy please squash your commits

**Comment by nsuchy** [comment]
Isn’t that what GitHub’s squash and merge is for? I can squash on my end if necessary although context on why would be nice. I have to attend a meeting (about 90 minutes) then I’ll take care of final touches to this PR :))

Cordially,
Nathaniel Suchy



Mar 6, 2019, 9:54 AM by notifications@github.com:

**Comment by nsuchy** [comment]
@vipulnsward Per your request the commits were squashed on my end.

**Comment by abhaynikam** [comment]
@sikachu @vipulnsward : Sorry for tagging on close PR. I can still see `http` links in some places like 
- https://github.com/rails/rails/blob/master/.github/pull_request_template.md
- https://github.com/rails/rails/blob/master/actioncable/actioncable.gemspec#L18

Is it fine if I update those to `https` and raise a PR?

**Comment by vipulnsward** [comment]
Sure, please do

**Pull request merged**